db/schema.rb

```diff
@@ -29,7 +28,7 @@
     t.string "record_type", null: false
     t.bigint "record_id", null: false
     t.bigint "blob_id", null: false
-    t.datetime "created_at", null: false
+    t.datetime "created_at", precision: nil, null: false
```

Comment:
all these `precision: nil` disappear from the file for me when I perform `db:reset`. Are you sure this schema is correct?

Reply:
For me they stay when running `db:reset`. It seems to have been added automatically when I added the room table.

To quote [this website](https://www.fastruby.io/blog/dealing-with-schema-changes-in-rails-7.html):

> Precision changes in Rails 7
> Rails now generates datetime columns with a default precision of 6. So that means t.datetime calls that don’t set the precision will have a default value of 6. 

> Before Rails 7, the default value of the precision was nil. So if your schema looked like this before Rails 7:
> `t.datetime "deleted_at"`
> Since this did not have any custom value of precision, it was internally being set to nil. Now when you upgrade to Rails 7, the default value is 6, so we need to explicitly set the precision: nil in the updated schema file. So the above line changes to:
> `t.datetime "deleted_at", precision: nil`

If I remove the `precision: nil` from the file and run `db:migrate` it adds them back again, to make our schema version 7 the same as version 6, where `nil` was default.

I can update the entire database to use the new default `precision: 6`, which would remove all the `precision` commands. 

I don't know why your localhost removes the precison comments. Im guessing your system still thinks you are on rails 6 instead of 7.

Reply:
okay, good to know! I'll make sure I am using an updated version of rails